documentation/cadl-structure-guidelines.md

```diff
@@ -0,0 +1,127 @@
+library, and to maximize the benefits of Cadl, this approach needs to be supported as a first-class citizen.
+
+## Repository
+
+Cadl can co-exist with Swagger within the existing `azure-rest-api-specs` repository. This approach will make it easier to generate Swagger artifacts without needing to sync between repos at the downside of having to live with any baggage associated with the repo as it ages. There are currently over 1,000 issues and 500 open PRs in this repo.
+
+## Structure Overview
+
+This proposal strives to align with [Azure SDK Repo Structure Guidelines](https://azure.github.io/azure-sdk/policies_repostructure.html).
+
+We eliminate the distinction between data-plane and resource-manager, treating all packages as siblings under the service family folder. This is what the SDKs do today. The legacy `data-plane` and `resource-manager` folders where Swagger specs live will remain.
+
+A given Cadl folder could represent any of the following scenarios:
+
+- an SDK and service spec (simplest, common for small services)
+- a service spec only
+- an SDK spec that aggregates many small service specs (one SDK to many microservices)
+- an SDK spec that extracts a portion of a large service spec (many SDKs to a single large service)
+- a shared library that represents neither a service spec nor an SDK
+
+Examples below:
+
+```
+-> specification
+   -> confidentialledger
+      -> ConfidentialLedger               (data-plane SDK + service)
+      -> data-plane                       (swagger)
+      -> resource-manager                 (swagger)
+   -> compute
+      -> Compute.Management               (mgmt SDK + service)
+      -> Compute.Management.Shared        (shared)
+      -> Compute.CloudService.Management  (service)
+      -> Compute.Diagnostic.Management    (service)
+      -> Compute.Disk.Management          (service)
+      -> Compute.Gallery.Management       (service)
+      -> Compute.Skus.Management          (service)
+      -> data-plane                       (swagger)
+      -> resource-manager                 (swagger)
+   -> keyvault
+      -> KeyVault.Certificates            (data-plane SDK + service)
+      -> KeyVault.Keys                    (data-plane SDK + service)
+      -> KeyVault.Secrets                 (data-plane SDK + service)
+      -> KeyVault.Management              (mgmt SDK + service)
+      -> data-plane                       (swagger)
+      -> resource-manager                 (swagger)
+```
+
+## Service Folders
+
+The service folder contains the entire Cadl library specification for a service, which could include custom linter rules, methods, etc.
+
+The folder name should correspond to the RP-name, but dropping the leading "Azure" or "Microsoft" prefix. Two additional naming conventions should be following:
+
+- `Management` at the end of the service RP-name indicates a management (resource manager) libarary.
```

Comment:
why management library needs a suffix, while data plane library does not?

Reply:
That is a fair question. Regardless we need some way to identify which is which and most of the time the mgmt sdk's often include the name management or resource manager in the names so that is what we chose here as well.